tests/beaker_tests/bgp/bgplib.rb

```diff
@@ -34,22 +34,36 @@ module BgpLib
   VRF2        = 'red'
 
   # Create manifest ensure => present + 'default' property values
-  def self.create_bgp_manifest_present
+  def self.create_bgp_manifest_present(platform, vrf='default')
+    conditional_props = ''
+    if platform != 'ios_xr'
+      conditional_props <<
+        "maxas_limit                            => 'default',
+         shutdown                               => 'default',
+         suppress_fib_pending                   => 'default',
+         bestpath_med_non_deterministic         => 'default',
+         timer_bestpath_limit                   => 'default',
+         timer_bestpath_limit_always            => 'default',
+         graceful_restart_helper                => 'default',
+         "
+    end
+
+    if vrf != 'default'
+      conditional_props <<
+        "route_distinguisher                    => 'default',"
+    end
+
     manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
     node 'default' {
       cisco_bgp { 'default':
         ensure                                 => present,
         asn                                    => #{BgpLib::ASN},
-        vrf                                    => 'default',
+        vrf                                    => '#{vrf}',
```

Comment:
There were three methods that were identical except for the vrf:  create_bgp_manifest_present, create_bgp_manifest_present_vrf1, and create_bgp_manifest_present_vrf2.  The intent of each was to test default values for global bgp, vrf1 ('blue') and vrf2 ('red'), respectively.  The last two methods now simply call the first method with 'blue' and 'red'.